packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx

```diff
@@ -32,51 +40,139 @@ class ArchiveAudioPlayer extends Component {
     }
 
-    if (Object.keys(stateUpdate).length > 0) {
-      this.setState(stateUpdate, postStateCB);
+    // This section is when Player does not start at first track
+    // Because of the way we initiate JWPlayer through IA's internal Play8.js wrapper,
+    //   we have to manually do some checks.
+    // This code is complementary to what is happening in `shouldComponentUpdate`
+    //   and `onPlaylistItemCB`
+    const playerStartingAtN = indexHasBeenSet && startAt === index && !playerPlaylistIndex;
+    const trackHasBeenSet = !Number.isInteger(prevPlayIndex) && Number.isInteger(index);
+
+    if (playerStartingAtN && !initiatePlay && trackHasBeenSet && !jwplayerOnloadStabilized) {
+      // JWPlayer has finally stabilized its initial onload through IA's internal wrapper: Play8.js
+      return this.setState({ jwplayerOnloadStabilized: true });
+    }
+
+    if (playerStartingAtN && !initiatePlay && jwplayerOnloadStabilized) {
+      // Once load has stabilized, we can confirm play management can start
+      return this.setState({ initiatePlay: true }, () => jwplayer(jwplayerID).pause());
     }
 
-    return null;
+    if (startAt === index && !playerPlaylistIndex && jwplayerOnloadStabilized && initiatePlay) {
+      // Manually play the track the player started on (N)
+      return this.setState({ playerPlaylistIndex: index }, () => jwplayer(jwplayerID).play());
+    }
   }
 
   /**
    * Event Handler that fires when JWPlayer starts a new track
    */
   onPlaylistItemCB(event) {
+    const {
+      initiatePlay,
+      onMetaFired,
+      playCBHasFired,
+    } = this.state;
+
+    if (!onMetaFired) return;
+
     const { index: playerPlaylistIndex } = event;
-    const { initiatePlay } = this.state;
-    const jwplayerStartingOnACertainTrack = !initiatePlay && (playerPlaylistIndex > 0);
-    if (initiatePlay || jwplayerStartingOnACertainTrack) {
+
+    if (initiatePlay) {
       this.emitPlaylistChange(playerPlaylistIndex);
     }
+
+    const jwplayerStartingOnACertainTrack = !initiatePlay && (playerPlaylistIndex > 0);
+    /*
+      This area checks when player loads on a track that isn't the 1st track
+      JWPlayer loads through IA's internal wrapper Play8.js
+      and this fires 2x (captured in `maxTimesPlayCBToBeFiredForLoad`).
+      We count callback (CB) firing when this happens and update state accordingly.
+
+      This code is complementary to `componentDidUpdate` and `shouldComponentUpdate`
+    */
+    if (jwplayerStartingOnACertainTrack && !initiatePlay) {
+      if (playCBHasFired === 0) {
+        // capture the index where the playlist starts & start count of CB fires
+        this.setState({ startAt: playerPlaylistIndex, playCBHasFired: playCBHasFired + 1 });
+      }
+
+      if (playCBHasFired === 1) {
+        // Once CB has fired once, we will continue to count the CB fire
+        // & emit that the player has updated index
+        this.setState({ playCBHasFired: playCBHasFired + 1 },
```

Comment:
Since this is only setting if `playCBHasFired` is 1, can we just set the new value to `2`?

Reply:
good suggestion. i will prefer to leave it as `+ 1` shows that we are counting.